### PR TITLE
teams: create the Xen Project Team and rename instances of `Xen` to `Xen Project`.

### PR DESCRIPTION
**Pull request by SigmaSquadron**
## Description of changes

- Creates the Xen Project Hypervisor Team, with **4** members who will maintain the Xen Project's ecosystem on Nixpkgs.

- Renames instances of "Xen" to "Xen Project" to comply with the Linux Foundation's and the Cloud Software Group's copyright agreements.

## Things done

- All parties have agreed to joining the team:
  - [x] Fernando Rodrigues
  - [x] HeHongBo
  - [x] Rane
  - [x] Yaroslav Bolyukin 
- Miscellaneous distro-wide changes:
  - [ ] The team page on the nixos.org website is up.
  - [x] A `6.topic: xen` label has been created.
  - [x] The `NixOS with Xen Project Hypervisor` matrix room has been created.
    - [x] The room's icon has been added to `nixos-artwork`. (Rename group to `Xen Project`)
    - [x] The room has been adopted by the `nixos.org` homeserver and space. (`#xen:nixos.org` is fine, like the option name is, as the room is clearly described as "Xen Project Hypervisor"-related.)
- All XPH-related packages are being maintained by the team:
  - [x] `xen` (pkgs/applications/virtualisation/xen)
    - Inherited by `qemu_xen`
  - [x] `xen-guest-agent`
  - [x] `xtf`
  - [x] `virtualisation.xen` (xen-dom0.nix)
  - [x] `virtualisation.xen.guest` (xen-domU.nix)
  - OCaml Packages:
    - [x] `ocamlPackages.xenstore`
    - [x] `ocamlPackages.xenstore_transport`
    - Inherited by `ocamlPackages.xenstore-tool`
    - [x] `ocamlPackages.vchan`
- [x] We have announced the creation of the team on Discourse and made an open invite for more maintainers/users.
- [x] The legal issues have been resolved.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

## E-mails

<details>
<summary><b>A Proposal for the Xen Maintenance Team for Nixpkgs</b></summary>

> Hello everyone!
>
> I'm Fernando, known as SigmaSquadron on GitHub/Matrix, and as you already know,
> I've been doing my best to revive Xen on NixOS. There is still much work to be
> done to fully integrate Xen with Nix's declarative nature, but I'm confident
> that we can achieve first-class Xen support by banding together as a team.
>
> Matei has been working on the DomU aspects of Xen, and I am currently preparing
> an accompanying module for the guest tools. Since the guest tools depend on the
> main Xen package, I believe we'd both benefit from tightly integrating these
> workflows, thereby making sure a Dom0 update doesn't break DomU tools, and
> vice-versa.
>
> Yaroslav has been working on the Lanzaboote boot builder for Dom0 Xen and
> hassingle-handedly packaged a POC Qubes dom0 on Nix. While the Qubes utilities
> are a bit out of scope for the Xen team, they'll certainly benefit from
> reliably-maintained dependencies. Maybe in the future we'll even be making a
> Qubes team!
>
> William, as the previous de facto Xen maintainer, your input will always be
> invaluable. I ask you to reconsider leaving and join us again, as you won't be
> maintaining the packages alone this time.
>
> I've also CC-ed other reviewers to the module PR; consider this an invitation to
> join the team as well! Although you're not currently listed as maintainers in
> Nixpkgs, you're familiar with the contribution process, and I hope you'll step
> up to help us co-maintain the Xen ecosystem. Since we're all together in the
> same team, none of us will be overwhelmed with Xen work, and we'll be able to
> more easily review each others' PRs to avoid regressions, as OfBorg will be
> notifying the whole team.
>
> If you're unaware, Nixpkgs has teams defined in maintainers/team-list.nix that
> include the purpose of the team and the list of members. Remember that
> maintaining a package doesn't mean rewriting it from scratch every time. Opening
> issues if other members of the team forget an update, testing PRs or fixing code
> quality/modernisation issues is already an immense help.
>
> The ultimate goal of the team is to distribute the maintainership responsibility
> and improve Xen's reliability on Nixpkgs. That said, now that the Constitution
> Draft has been published, I was also hoping to ask the Steering Committee for a
> @nixos.org email specifically meant to receive embargoed XSAs. We had trouble
> patching Xen in the last few XSA releases, as upstream Xen forgot to make the
> patches public in a timely manner. Being a part of the pre-disclosure list means
> that this is no longer an issue, as we can prepare vendored patches and merge
> them in minutes after the embargo ends, not days. Having a whole team behind Xen
> would officialise NixOS's position as a distributor of the Hypervisor.
>
> Thank you for reading, and I hope you will all accept this invitation to start a
> Xen team. If you do, please include any non-email contact information you use
> for interacting with NixOS, such as a Matrix/NixOS Discourse handle, in your
> reply.
>
> Appreciatively yours,<br>
> Fernando Rodrigues.
>
> E3CD E225 47C6 2DB6 6CCD BC06 CC3A E2EA 0000 0000
</details>

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -17548,6 +17554,12 @@
     githubId = 8099415;
     name = "Raymond Douglas";
   };
+  rayne = {
+    email = "rayne+nix@digitalrayne.net";
+    github = "digitalrayne";
+    githubId = 1829286;
+    name = "Rayne";
+  };
```
Comment (digitalrane) [inline maintainers/maintainer-list.nix:17562]:
```suggestion
  rane = {
    email = "rane+nix@junkyard.systems";
    github = "digitalrane";
    githubId = 1829286;
    name = "Rane";
  };
```

File: maintainers/team-list.nix [hunk 2]
```diff
@@ -1022,6 +1022,18 @@ with lib.maintainers;
     shortName = "WDZ GmbH";
   };
 
+  xen = {
+    members = [
+      hehongbo
+      lach
+      rayne
```
Comment (digitalrane) [inline maintainers/team-list.nix:1029]:
```suggestion
      rane
```

**Review by digitalrane — COMMENTED** [review]
Just a couple of changes, thanks for putting this together @SigmaSquadron  😄

**Review by CertainLach — APPROVED** [review]

**Review by hehongbo — APPROVED** [review]
Looks good and thanks!

**Review by ulrikstrid — CHANGES_REQUESTED** [review]
Looks good overall, just a small cleanup then it should be good to go :)

File: pkgs/development/ocaml-modules/xenstore/default.nix [hunk 3]
```diff
@@ -25,10 +25,7 @@ buildDunePackage rec {
   meta = with lib; {
     description = "Xenstore protocol in pure OCaml";
     license = licenses.lgpl21Only;
-    maintainers = with maintainers; [
-      sternenseemann
-      sigmasquadron
-    ];
+    maintainers = teams.xen.members ++ (with maintainers; [ sternenseemann ]);
```
[the remark this answers is not in the document]
Comment (SigmaSquadron) [reply pkgs/development/ocaml-modules/xenstore/default.nix]: Done!

**Review by hehongbo — APPROVED** [review]

**Review by CertainLach — APPROVED** [review]

**Review by digitalrane — APPROVED** [review]

**Comment by SigmaSquadron** [comment]
oh I just noticed that OfBorg probably tried to build Microsoft Teams due to the `teams:` tag in the PR.

**Review by ulrikstrid — APPROVED** [review]

**Review by emilazy — APPROVED** [review]

**Pull request merged**